Core/src/main/java/com/plotsquared/core/util/TabCompletions.java

```diff
@@ -222,9 +262,28 @@ private TabCompletions() {
      * @param existing        Players that should not be included in completions
      * @param uuidFilter      Filter applied before caching values
      * @return List of completions
+     * @deprecated In favor {@link #completePlayers(String, PlotPlayer, String, List, Predicate)}
+     */
+    @Deprecated(forRemoval = true)
+    private static List<Command> completePlayers(
+            final @NonNull String cacheIdentifier,
+            final @NonNull String input, final @NonNull List<String> existing,
+            final @NonNull Predicate<UUID> uuidFilter
+    ) {
+        return completePlayers(cacheIdentifier, null, input, existing, uuidFilter);
+    }
+
+    /**
+     * @param cacheIdentifier Cache key
+     * @param issuer          The player who issued the tab completion
+     * @param input           Command input
+     * @param existing        Players that should not be included in completions
+     * @param uuidFilter      Filter applied before caching values
+     * @return List of completions
      */
     private static List<Command> completePlayers(
             final @NonNull String cacheIdentifier,
+            final @Nullable PlotPlayer<?> issuer,
```

Comment:
Would it make sense to make the issuer NonNull too and use the console (`ConsolePlayer.getConsole()`) as default in the old methods instead?

Reply:
Sure, I can do that. It doesn't change the logic behind it, it's just a pure API design choice. I don't know how you handle this.

Reply:
Commands are always executed with an issuer, but making it Nullable could indicate otherwise. So yeah, I'd prefer to make it NonNull